faiss/impl/HNSW.cpp

```diff
@@ -7,11 +7,18 @@
 
 // -*- c++ -*-
 
-#include <faiss/impl/HNSW.h>
-
+#include <mmintrin.h>
+#include <xmmintrin.h>
 #include <string>
 
 #include <faiss/impl/AuxIndexStructures.h>
+#include <faiss/impl/HNSW.h>
+
+#ifdef __SSE__
+#define MM_PREFETCH(a) _mm_prefetch((a), _MM_HINT_T0)
```

Comment:
Could you call it a different name so that it is not too x86 centric? Like PREFETCH_HINT or something?